lib/post_creator.rb

```diff
@@ -128,6 +128,23 @@ def valid?
         errors.add(:base, I18n.t(:not_accepting_pms, username: users[m]))
       end
 
+      # Is Allowed PM users list enabled for any recipients?
+      users_with_allowed_pms = allowed_pms_enabled(users).pluck(:id).uniq
+
+      # If any of the users has allowed_pm_users enabled check to see if the creator
+      # is in their list
+      if users_with_allowed_pms.any?
+        users_sender_can_pm = allowed_pms_enabled(users)
+          .where("allowed_pm_users.allowed_pm_user_id = #{@user.id.to_i}")
```

Comment:
```suggestion
          .where("allowed_pm_users.allowed_pm_user_id" => @user.id.to_i)
```
I *think* the above syntax works and means you needn't bother with string interpolation which is riskier.